Diffusion/Common/ImportModelBehavior.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  ImportModelBehavior.swift
+//  Diffusion-macOS
+//
+//  Created by Dolmere on 19/06/2023
+//  See LICENSE at https://github.com/huggingface/swift-coreml-diffusers/LICENSE
+
+
+import SwiftUI
+import ZIPFoundation
+
+// A view modifier to add a .fileImporter import panel into a view
+struct ImportModelBehavior: ViewModifier {
+
+    @ObservedObject private var settings = Settings.shared
+
+    @State var isBadSelectionAlertShown: Bool = false
+    @State private var importPanelState: Bool = false {
+        didSet {
+            if settings.isShowingImportPanel != importPanelState {
+                settings.isShowingImportPanel = importPanelState
+            }
+        }
+    }
+
+    func body(content: Content) -> some View {
+        content
+            .fileImporter(isPresented: $importPanelState, allowedContentTypes: [.folder, .zip]) { result in
+                do {
+                    let url = try result.get()
+                    if url.hasDirectoryPath {
+                        let fileManager = FileManager.default
+                        let contents = try fileManager.contentsOfDirectory(atPath: url.path)
+                        
+                        if contents.contains(where: { $0.caseInsensitiveCompare("merges.txt") == .orderedSame }) && contents.contains(where: { $0.caseInsensitiveCompare("vocab.json") == .orderedSame }) {
+                            // Folder contains the required filenames
+                            do {
+                                try fileManager.moveItem(at: url, to: Settings.shared.applicationSupportURL().appendingPathComponent("hf-diffusion-models").appendingPathComponent(url.lastPathComponent))
+                            } catch {
+                                // Error handling
+                                print("Error: \(error.localizedDescription)")
+                            }
+                        } else {
+                            // Folder does not contain the required filenames
+                            isBadSelectionAlertShown = true
+                        }
+                    } else if url.pathExtension == "zip" {
+                        DispatchQueue.global(qos: .background).async {
+                            do {
+                                let extractedURL = Settings.shared.applicationSupportURL().appendingPathComponent("hf-diffusion-models").appendingPathComponent(url.deletingPathExtension().lastPathComponent)
```

Comment:
Maybe make `PipelineLoader.models` public somewhere, so we make sure it's always in sync?

Reply:
`PipelineLoader.models` could best be moved to Settings (Settings.shared.models) or, as we will come to later with more extractions from #51, a new observable object like "ModelsViewModel."

Reply:
Maybe we should merge #60 first?

Reply:
Happy to bring #60 in and resolve it into this PR. Would you please test on your side the behaviour you see vs. the shipping version before accepting? I've been around the houses on what I've seen and just want to be reassured I'm not introducing a bug due to my own local config.